app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/hplus/HPlusCoordinator.java

```diff
@@ -0,0 +1,236 @@
+
+    @Override
+    public boolean supportsActivityTracking() {
+        return true;
+    }
+
+    @Override
+    public SampleProvider<? extends ActivitySample> getSampleProvider(GBDevice device, DaoSession session) {
+        return new HPlusSampleProvider(device, session);
+    }
+
+    @Override
+    public boolean supportsScreenshots() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsAlarmConfiguration() {
+        return true;
+    }
+
+    @Override
+    public boolean supportsHeartRateMeasurement(GBDevice device) {
+        return true;
+    }
+
+    @Override
+    public int getTapString() {
+        return R.string.tap_connected_device_for_activity;
+    }
+
+    @Override
+    public String getManufacturer() {
+        return "Zeblaze";
+    }
+
+    @Override
+    public boolean supportsAppsManagement() {
+        return false;
+    }
+
+    @Override
+    public Class<? extends Activity> getAppsManagementActivity() {
+        return null;
+    }
+
+    @Override
+    protected void deleteDevice(@NonNull GBDevice gbDevice, @NonNull Device device, @NonNull DaoSession session) throws GBException {
+        // nothing to delete, yet
+    }
+
+    public static int getFitnessGoal(String address) throws IllegalArgumentException {
+        Prefs prefs = GBApplication.getPrefs();
+        return prefs.getInt(HPlusConstants.PREF_HPLUS_FITNESS_GOAL + "_" + address, 10000);
+    }
+
+    /**
+     * Returns the user info from the user configured data in the preferences.
+     *
+     * @param hplusAddress
+     * @throws IllegalArgumentException when the user info can not be created
+     */
+    public static UserInfo getConfiguredUserInfo(String hplusAddress) throws IllegalArgumentException {
+        ActivityUser activityUser = new ActivityUser();
+
+        UserInfo info = UserInfo.create(
+                hplusAddress,
+                prefs.getString(HPlusConstants.PREF_HPLUS_USER_ALIAS, null),
+                activityUser.getGender(),
+                activityUser.getAge(),
+                activityUser.getHeightCm(),
+                activityUser.getWeightKg(),
+                0
+        );
+        return info;
+    }
+
+    public static byte getCountry(String address) {
+        return (byte) prefs.getInt(HPlusConstants.PREF_HPLUS_COUNTRY + "_" + address, 10);
+
+    }
+
+    public static byte getTimeMode(String address) {
+        return (byte) prefs.getInt(HPlusConstants.PREF_HPLUS_TIMEMODE + "_" + address, 0);
+    }
+
+    public static byte getUnit(String address) {
+        return (byte) prefs.getInt(HPlusConstants.PREF_HPLUS_UNIT + "_" + address, 0);
+    }
+
+    public static byte getUserWeight(String address) {
+        ActivityUser activityUser = new ActivityUser();
+
+        return (byte) activityUser.getWeightKg();
+    }
+
+    public static byte getUserHeight(String address) {
+        ActivityUser activityUser = new ActivityUser();
+
+        return (byte) activityUser.getHeightCm();
```

Comment:
Here you might need to ensure unsigned value with & 0xff. You basically need to do this everywhere a value > 128 is used.